### PR TITLE
[CJRSC-225] Phase Two Calculations

### PR DESCRIPTION
**Pull request by jordanrowe**

### CODE REVIEW

File: app/controllers/ConfirmationController.scala [hunk 1]
```diff
@@ -56,6 +58,9 @@ class ConfirmationController @Inject()(
         } else {
           Future.successful(Ok(view(data.metaData, data.confirmationViewBreakdown, config.calculatorVersion)))
         }
+      case Valid(data: PhaseTwoConfirmationDataResult) =>
+        Future.successful(Ok(phaseTwoView(data.confirmationViewBreakdown, data.metaData.claimPeriod, config.calculatorVersion)))
```
Comment (sureshhmrc) [inline app/controllers/ConfirmationController.scala:62]: what happens to `auditing` incase of `PhaseTwo` ?

Comment (jordanrowe) [reply app/controllers/ConfirmationController.scala:62]: will be implemented under a different ticket, CJRSC-253

File: app/models/PeriodBreakdown.scala [hunk 2]
```diff
@@ -110,18 +112,89 @@ sealed trait PhaseTwoPeriodBreakdown {
...
+  def isPartial = paymentWithPeriod.phaseTwoPeriod.periodWithPaymentDate.period.isInstanceOf[PartialPeriod]
+  def isPartTime = paymentWithPeriod.phaseTwoPeriod.isPartTime
+
+  def thresholdMessage(implicit messages: Messages): String =
+    (isPartial, isPartTime) match {
+      case (false, false) => messages("phaseTwoNicBreakdown.l3", threshold.value.formatted("%.2f"))
+      case (true, false) =>
+        messages(
+          "phaseTwoNicBreakdown.l3.partial",
+          threshold.value.formatted("%.2f"),
```
Comment (sureshhmrc) [inline app/models/PeriodBreakdown.scala:135]: minor :  we can have an implicit or something  - with out using `.formatted("%.2f")` explicitly several times  -  but not a big deal we can leave it for now

File: test/handlers/ConfirmationControllerRequestHandlerSpec.scala [hunk 3]
```diff
@@ -237,4 +237,8 @@ class ConfirmationControllerRequestHandlerSpec extends SpecBase with CoreTestDat
     (loadResultData(manyPeriods).value.asInstanceOf[PhaseOneConfirmationDataResult]).confirmationViewBreakdown.furlough.total mustBe 2402.63
   }
 
+  "do phase two calculation if claimStartDate is on or after 1 July 2020" in new ConfirmationControllerRequestHandler {
+    loadResultData(phaseTwoJourney()).value.isInstanceOf[PhaseTwoConfirmationDataResult] mustBe true
```
Comment (chrisjameswright) [inline test/handlers/ConfirmationControllerRequestHandlerSpec.scala:241]: Using a matcher for the type should give better error messages were this to fail:
```suggestion
    loadResultData(phaseTwoJourney()).value mustBe a[PhaseTwoConfirmationDataResult]
```

**Pull request merged**